tensorflow_datasets/image/cityscapes.py

```diff
@@ -0,0 +1,280 @@
+
+  def _split_generators(self, dl_manager):
+    paths = {}
+    for split, (zip_file, zip_root) in self.builder_config.zip_root.items():
+      paths[split] = os.path.join(dl_manager.manual_dir, zip_file)
+
+    if any(not os.path.exists(z) for z in paths.values()):
+      msg = 'You must download the dataset files manually and place them in: '
+      msg += ', '.join(paths.values())
+      raise AssertionError(msg)
+
+    for split, (_, zip_root) in self.builder_config.zip_root.items():
+      paths[split] = os.path.join(dl_manager.extract(paths[split]), zip_root)
+
+    splits = [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                feat_dir: os.path.join(path, 'train')
+                for feat_dir, path in paths.items()
+                if not feat_dir.endswith('/extra')
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            gen_kwargs={
+                feat_dir: os.path.join(path, 'val')
+                for feat_dir, path in paths.items()
+                if not feat_dir.endswith('/extra')
+            },
+        ),
+    ]
+
+    # Test split does not exist in coarse dataset
+    if not self.builder_config.train_extra_split:
+      splits.append(tfds.core.SplitGenerator(
+          name=tfds.Split.TEST,
+          gen_kwargs={
+              feat_dir: os.path.join(path, 'test')
+              for feat_dir, path in paths.items()
+              if not feat_dir.endswith('/extra')
+          },
+      ))
+    else:
+      splits.append(tfds.core.SplitGenerator(
+          name='train_extra',
+          gen_kwargs={
+              feat_dir.replace('/extra', ''): os.path.join(path, 'train_extra')
+              for feat_dir, path in paths.items()
+              if feat_dir.endswith('/extra')
+          },
+      ))
+    return splits
+
+  def _generate_examples(self, **paths):
+    left_imgs_root = paths['images_left']
+    for city_id in tf.io.gfile.listdir(left_imgs_root):
+      paths_city_root = {feat_dir: os.path.join(path, city_id)
+                         for feat_dir, path in paths.items()}
+
+      left_city_root = paths_city_root['images_left']
+      for left_img in tf.io.gfile.listdir(left_city_root):
+        left_img_path = os.path.join(left_city_root, left_img)
+        image_id = _get_left_image_id(left_img)
+
+        if image_id in self.builder_config.ignored_ids:
+          continue
+
+        features = {
+            'image_id': image_id,
+            'image_left': left_img_path
+        }
+
+        if self.builder_config.right_images:
+          features['image_right'] = os.path.join(
+              paths_city_root['images_right'], f'{image_id}_rightImg8bit.png')
+
+        if self.builder_config.segmentation_labels:
+          features['segmentation_label'] = os.path.join(
+              paths_city_root['segmentation_labels'],
+              f'{image_id}_{self.builder_config.label_suffix}.png')
```

Comment:
This is using python3 only feature, while TFDS should be both Py2 and 3 compatible.

Reply:
@f1recracker https://github.com/f1recracker/tensorflow-datasets/pull/1 here is a possible fix

Reply:
will fix tonight. Thanks